### PR TITLE
added API Endpoint to create group

### PR DESCRIPTION
**Pull request by saurabhjn76**
Fixes issue #293 
Changes: added CreateGroups.java endpoint for creating groups
Screenshots for the change: 
image
image
The endpoint can be tested [http://127.0.0.1:4000/aaa/createGroups.json?group=groupName&role=admin](http://127.0.0.1:4000/aaa/createGroups.json?group=groupName&role=admin)

### CODE REVIEW

**Comment by saurabhjn76** [comment]
@dynamitechetan @jyoth1raditya @madhavrathi  please review

File: src/ai/susi/server/api/aaa/CreateGroups.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Override
+    public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
+        return null;
+    }
+
+    @Override
+    public String getAPIPath() {
+        return "/cms/createGroups.json";
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/CreateGroups.java]: This servlet is in the aaa Package and therefore the path should be "/aaa/createGroup.json";

not "groups". It creates one group. Therefore "group"

File: src/ai/susi/server/api/aaa/CreateGroups.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.File;
+
+/**
+ * This Servlets creates a group with default baserole as admin.
+ * It takes 1 parameter group and other optional parameter role.
+ * Can be tested on
+ * http://127.0.0.1:4000/aaa/createGroups.json?group=groupName&role=admin
+ *
+ */
+public class CreateGroups extends AbstractAPIHandler implements APIHandler {
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/CreateGroups.java]: The naming schema is using "service" as name part. And it should be singular as it creates one group, not several groups. Therefore the name should be "CreateGroupService"

File: src/ai/susi/server/api/aaa/CreateGroups.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    @Override
+    public String getAPIPath() {
+        return "/cms/createGroups.json";
+    }
+
+    @Override
+    public ServiceResponse serviceImpl(Query call, HttpServletResponse response, Authorization rights, final JsonObjectWithDefault permissions) {
+        JSONObject result = new JSONObject();
+        String groupName = call.get("group", null);
+        String grouppBaseRole = call.get("role","admin");
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/CreateGroups.java]: having "admin" as default is dangerous. Please use a default group with  low rights.

**Comment by saurabhjn76** [comment]
@Orbiter  made the changes.

**Comment by saurabhjn76** [comment]
@Orbiter removed Unnecessary import statements

**Pull request merged**